app.py

```diff
@@ -66,29 +75,50 @@ def snap_details(snap_name):
     some of the data through to the snap-details.html template,
     with appropriate sanitation.
     """
+    today = datetime.date.today()
+    month_ago = today - relativedelta.relativedelta(months=1)
 
-    query_headers = {
-        'X-Ubuntu-Series': '16',
-        'X-Ubuntu-Architecture': 'amd64',
-    }
-
-    response = _get_from_cache(
+    details_response = _get_from_cache(
         snap_details_url.format(snap_name=snap_name),
-        headers=query_headers
+        headers=details_query_headers
+    )
+    details = details_response.json()
+
+    metrics_query_json = [
+        {
+            "metric_name": "installed_base_by_country_percent",
+            "snap_id": details['snap_id'],
+            "start": month_ago.strftime('%Y-%m-%d'),
+            "end": today.strftime('%Y-%m-%d')
+        }
+    ]
+    metrics_response = _get_from_cache(
+        snap_metrics_url.format(snap_name=snap_name),
+        headers=metrics_query_headers,
+        json=metrics_query_json
     )
+    geodata = metrics_response.json()[0]['series']
 
-    if response.status_code >= 400:
+    user_percentage_by_country = {}
+
+    for country_percentages in geodata:
+        country_code = country_percentages['name']
+        percentages_with_nulls = country_percentages['values']
+        percentages = [p for p in percentages_with_nulls if p is not None]
+        average_percentage = sum(percentages) / len(percentages)
+        user_percentage_by_country[country_code] = average_percentage
```

Comment:
What is the rationale for averaging last 30d of installed_base ? I am afraid we are diluting meaning unnecessarily.

Reply:
To start with we'd like to implement a simple solution so we can ship it. So rather than offering a map that can be filtered by date, we'll simply show an up-to-date snapshot - so we're only going to show a single data point per country.

In discussion with @matthewpaulthomas we decided that simply showing the *mos recent* day of data might be too subject to anomalous fluctuations (people's computers were switched on), so it's better to take an average over a larger period. We chose a month, but if you think a different period makes sense then let me know.

Reply:
I discussed the question of the this smoothing over a period of time with @pmahnke and he agreed that, for the time being, taking the average over the last month seems like it should probably produce fresh-enough data. So I'm going to merge it as it is currently implemented. However, we can continue to discuss and refine this implementation.